modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/MavenDeployableProjectModelBuilder.java

```diff
@@ -159,36 +170,47 @@ private Supplier<MuleDeployableModel> getDeployableModelResolver(ArtifactCoordin
 
   private boolean shouldEditApplicationModel(MuleApplicationModel applicationModel) {
     return (exportAllResourcesAndPackagesIfEmptyLoaderDescriptor
-        && applicationModel.getClassLoaderModelLoaderDescriptor() == null) || includeTestDependencies;
+        && applicationModel.getClassLoaderModelLoaderDescriptor() == null) || includeTestDependencies
+        || applicationModel.getConfigs() == null;
   }
 
   @Override
   protected boolean isIncludeTestDependencies() {
     return includeTestDependencies;
   }
 
-  private MuleApplicationModel buildApplicationModel(MuleApplicationModel applicationModel, List<String> resources,
+  private MuleApplicationModel buildApplicationModel(MuleApplicationModel applicationModel, List<String> allResources,
+                                                     String muleResourcesDirectory, List<String> muleResources,
                                                      List<String> packages) {
     MuleApplicationModel.MuleApplicationModelBuilder builder = new MuleApplicationModel.MuleApplicationModelBuilder()
         .setName(applicationModel.getName() != null ? applicationModel.getName() : "mule")
         .setMinMuleVersion(applicationModel.getMinMuleVersion())
         .setRequiredProduct(applicationModel.getRequiredProduct())
         .withClassLoaderModelDescriptorLoader(createClassLoaderModelDescriptorLoader(applicationModel
-            .getClassLoaderModelLoaderDescriptor(), resources, packages))
+            .getClassLoaderModelLoaderDescriptor(), allResources, packages))
```

Comment:
do the concat between `resources` and `muleResources` here instead of propagating the `allResources` parameter everywhere.

Reply:
I kept the `allResources` there since now I no longer need to propagate the `muleResources`.